core/src/main/kotlin/com/justai/jaicf/activator/StateMapActivator.kt

```diff
@@ -1,47 +1,88 @@
 package com.justai.jaicf.activator
 
+import com.justai.jaicf.api.BotRequest
+import com.justai.jaicf.context.ActivatorContext
 import com.justai.jaicf.context.BotContext
-import com.justai.jaicf.model.activation.ActivationRuleType
+import com.justai.jaicf.model.activation.Activation
+import com.justai.jaicf.model.activation.ActivationRule
 import com.justai.jaicf.model.scenario.ScenarioModel
 import com.justai.jaicf.model.state.StatePath
+import com.justai.jaicf.model.transition.Transition
 
 /**
  * A helper abstraction for every [Activator] that should activate a state if it contains a particular rule like event or intent.
  *
  * @see com.justai.jaicf.activator.event.BaseEventActivator
  * @see com.justai.jaicf.activator.intent.BaseIntentActivator
  */
-abstract class StateMapActivator(
-    ruleType: ActivationRuleType,
-    model: ScenarioModel
-): Activator {
-
-    private val transitions = model.activations
-        .filter { a -> a.type == ruleType }
-        .map { a -> Pair(a.fromState, Pair(a.rule, a.toState)) }
-        .groupBy {a -> a.first}
-        .mapValues { l -> l.value.map { v -> v.second } }
-
-    fun findState(rule: String?, botContext: BotContext): String? = rule?.let {
-        val path = StatePath.parse(botContext.dialogContext.currentContext)
-        return checkWithParents(path, rule)
+abstract class StateMapActivator(model: ScenarioModel) : Activator {
```

Comment:
Maybe it's time to rename this class to `BaseActivator` (or something like that) once it's a single direct implementation of `Activator`?